code/go/0chain.net/blobbercore/reference/ref.go

```diff
@@ -225,6 +224,109 @@ func GetReference(ctx context.Context, allocationID, path string) (*Ref, error)
 	return nil, err
 }
 
+// GetReferenceID get FileRef ID with allcationID and path from postgres
+func GetReferenceID(ctx context.Context, allocationID, path string) (*Ref, error) {
+	ref := &Ref{}
+	db := datastore.GetStore().GetTransaction(ctx)
+	db = db.Select("id")
+	err := db.Where(&Ref{AllocationID: allocationID, Path: path}).First(ref).Error
+	if err == nil {
+		return ref, nil
+	}
+	return nil, err
+}
+
+// GetReferenceFileUpdate get FileRef with allcationID and path from postgres
+func GetReferenceFileUpdate(ctx context.Context, allocationID, path string) (*Ref, error) {
+	ref := &Ref{}
+	db := datastore.GetStore().GetTransaction(ctx)
+	db = db.Select("id", "on_cloud", "size")
+	err := db.Where(&Ref{AllocationID: allocationID, Path: path}).First(ref).Error
+	if err == nil {
+		return ref, nil
+	}
+	return nil, err
+}
+
+// GetReferenceHash get FileRef ID with allcationID and path from postgres
+func GetReferenceHash(ctx context.Context, allocationID, path string) (*Ref, error) {
```

Comment:
If `hash` is only required then its better to return `hash` rather than `ref`. decoding to `ref` is costlier